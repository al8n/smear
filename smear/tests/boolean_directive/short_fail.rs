#[derive(smear::BooleanDirective)]
#[smear(short = '9')]
struct NumberChar;

#[derive(smear::BooleanDirective)]
#[smear(short = '_')]
struct UnderscoreChar;

#[derive(smear::BooleanDirective)]
#[smear(short = '-')]
struct InvalidChar;

#[derive(smear::BooleanDirective)]
#[smear(short = 9)]
struct InvalidLit;

#[derive(smear::BooleanDirective)]
#[smear(short())]
struct InvalidMeta;

fn main() {}