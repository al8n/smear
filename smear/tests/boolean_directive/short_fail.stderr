error: The value of `short` must be an ASCII alphabetic character, but got '9'.
 --> tests/boolean_directive/short_fail.rs:2:17
  |
2 | #[smear(short = '9')]
  |                 ^^^

error: The value of `short` must be an ASCII alphabetic character, but got '_'.
 --> tests/boolean_directive/short_fail.rs:6:17
  |
6 | #[smear(short = '_')]
  |                 ^^^

error: The value of `short` must be an ASCII alphabetic character, but got '-'.
  --> tests/boolean_directive/short_fail.rs:10:17
   |
10 | #[smear(short = '-')]
   |                 ^^^

error: Unexpected literal type `int`
  --> tests/boolean_directive/short_fail.rs:14:17
   |
14 | #[smear(short = 9)]
   |                 ^

error: expected a value for this attribute: `short` or `short = '...'`
  --> tests/boolean_directive/short_fail.rs:18:9
   |
18 | #[smear(short())]
   |         ^^^^^^^
